docs/ops/detection/ROIAlign_9.md

```diff
@@ -0,0 +1,117 @@
+# ROIAlign {#openvino_docs_ops_detection_ROIAlign_9}
+
+**Versioned name**: *ROIAlign-9*
+
+**Category**: *Object detection*
+
+**Short description**: *ROIAlign* is a *pooling layer* used over feature maps of non-uniform input sizes and outputs a feature map of a fixed size.
+
+**Detailed description**: [Reference](https://arxiv.org/abs/1703.06870).
+
+*ROIAlign* performs the following for each Region of Interest (ROI) for each input feature map:
+1. Multiply box coordinates with *spatial_scale* to produce box coordinates relative to the input feature map size based on *aligned_mode* attribute.
+2. Divide the box into bins according to the *sampling_ratio* attribute.
+3. Apply bilinear interpolation with 4 points in each bin and apply maximum or average pooling based on *mode* attribute to produce output feature map element.
+
+**Attributes**
+
+* *pooled_h*
+
+  * **Description**: *pooled_h* is the height of the ROI output feature map.
+  * **Range of values**: a positive integer
+  * **Type**: `int`
+  * **Required**: *yes*
+
+* *pooled_w*
+
+  * **Description**: *pooled_w* is the width of the ROI output feature map.
+  * **Range of values**: a positive integer
+  * **Type**: `int`
+  * **Required**: *yes*
+
+* *sampling_ratio*
+
+  * **Description**: *sampling_ratio* is the number of bins over height and width to use to calculate each output feature map element. If the value
+  is equal to 0 then use adaptive number of elements over height and width: `ceil(roi_height / pooled_h)` and `ceil(roi_width / pooled_w)` respectively.
+  * **Range of values**: a non-negative integer
+  * **Type**: `int`
+  * **Required**: *yes*
+
+* *spatial_scale*
+
+  * **Description**: *spatial_scale* is a multiplicative spatial scale factor to translate ROI coordinates from their input spatial scale to the scale used when pooling.
+  * **Range of values**: a positive floating-point number
+  * **Type**: `float`
+  * **Required**: *yes*
+
+* *mode*
+
+  * **Description**: *mode* specifies a method to perform pooling to produce output feature map elements.
+  * **Range of values**:
+    * *max* - maximum pooling
+    * *avg* - average pooling
+  * **Type**: string
+  * **Required**: *yes*
+
+  * *aligned_mode*
+
+  * **Description**: *aligned_mode* specifies how to transform the coordinate in original tensor to the resized tensor.
+  * **Range of values**: name of the transformation mode in string format (here scale[x] is resized_shape[x] / original_shape[x] and x_original is a coordinate in axis x, for any axis x from the input axes):
+    * *asymmetric* - the coordinate in the resized tensor axis x is calculated according to the formula x_original * scale[x]
+    * *half_pixel* - the coordinate in the resized tensor axis x is calculated as ((x_original + 0.5) * scale[x]) - 0.5
+    * *half_pixel_for_nn* - the coordinate in the resized tensor axis x is x_original * scale[x] - 0.5
```

Comment:
why do we have three modes if PDPD and PyTorch just have two modes?

Reply:
Hi, Roman: 
'asymmetric' mode can be considered as coordinates transformed without alignment.
'half_pixel' and 'half_pixel_for_nn' can be considered as two kinds of half_pixel coordinates transformed way for alignment. and these two kinds of coordinates transformed way are **very similar(and easily to be changed into each other during usage**), the main difference for these two ways is whether the original tensor coordinates have already included the 'offset'(0.5), if user have included this 'offset' in original input tensor(e.g. Paddle FasterRCNN model default way), it will use 'half_pixel_for_nn' mode, if user not included this 'offset' in original input tensor, it will use 'half_pixel' mode(the complete formula form of half pixel coordinates transformed way for alignment).

Reply:
I think we need to use two modes as described in PDPD and PyTorch (legacy, alignment with the two neighboring pixel indices). This alignment with FWs will help us with clarity in the future.
As I understand, `asymmetric` - legacy in PDPD, PyTorch and `half_pixel_for_nn` - for alignment with the two neighboring pixel indices
`half_pixel` can be implemented as adding 0.5 constant prior to this `ROIAlign`.
Let us proceed with two modes.

Reply:
Hi, Roman:
agree that two modes('asymmetric' and 'half_pixel_for_nn') is enough for our current model support target. but I still suggest that we added 'half_pixel' mode for 'ROIAlign_9' for the following two reasons:

(1)Some Framework(e.g.Paddle) models used 'half_pixel_for_nn' for roialign because it merged '0.5 constant offset process'(some kind of postprocess) in prior ops of roialign. while in the future if this offset process can't be merged into prior ops anymore(maybe caused by graph optimization methods changed), then it will use 'half_pixel' mode.

(2)yes, as you mentioned. we indeed can add 0.5 offset data movement prior to this ROIAlign in our op conversion in case to support 'half_pixel' mode. While I think these pixel by pixel data movement will increase time coast of this op.

so, it looks like additionally supporting 'half_pixel' mode will takes us very little extra effort but gain benefits(or insurance) for the future needed.